pallets/parachain-staking/src/tests/schedule_unbond_tests.rs

```diff
@@ -0,0 +1,183 @@
+                }
+            })
+            .filter_map(|inner| {
+                if let Event::NominationDecreaseScheduled { nominator, .. } = inner {
+                    Some(nominator)
+                } else {
+                    None
+                }
+            })
+            .any(|n| n == nominator)
+    }
+
+    #[test]
+    fn unbond_test() {
+        ExtBuilder::default().build().execute_with(|| {
+            let num_collators = 10;
+
+            let mut nominations: Vec<u128> = vec![7, 7, 7, 7, 7, 7, 7, 7, 7, 7];
+            for i in 1..=get_max_to_unbond(&nominations, num_collators) {
+                AMOUNT_TO_UNBOND.with(|pk| *pk.borrow_mut() = i);
+                unbond(num_collators.clone(), &nominations);
+            }
+
+            nominations = vec![6, 6, 6, 6, 6, 6, 6, 6, 6, 11];
+            for i in 1..=get_max_to_unbond(&nominations, num_collators) {
+                AMOUNT_TO_UNBOND.with(|pk| *pk.borrow_mut() = i);
+                unbond(num_collators.clone(), &nominations);
+            }
+
+            nominations = vec![15, 11, 11, 11, 13, 19, 19, 10, 10, 14];
+            for i in 1..=get_max_to_unbond(&nominations, num_collators) {
+                AMOUNT_TO_UNBOND.with(|pk| *pk.borrow_mut() = i);
+                unbond(num_collators.clone(), &nominations);
+            }
+
+            nominations = vec![102, 4, 13, 25, 21, 3, 49, 11, 39, 87];
+            for i in 1..=get_max_to_unbond(&nominations, num_collators) {
+                AMOUNT_TO_UNBOND.with(|pk| *pk.borrow_mut() = i);
+                unbond(num_collators.clone(), &nominations);
+            }
+        });
+    }
+
+    fn unbond(num_collators: u32, nominations: &Vec<u128>) {
+        let initial_collator_stake = 10;
+        let initial_balance = 1000000000000000000000;
+        let staker: Staker = Default::default();
+        let staker_balance =
+            vec![(staker.account_id, initial_balance), (staker.relayer, initial_balance)];
+        ExtBuilder::default()
+            .with_balances(get_accounts(num_collators, initial_balance, Some(staker_balance)))
+            .with_candidates(get_accounts(num_collators, initial_collator_stake, None))
+            .with_nominations(get_nominations(num_collators, staker.account_id, nominations))
+            .with_staking_config(
+                initial_collator_stake,
+                MinNominationPerCollator::get() * num_collators as u128,
+            )
+            .build()
+            .execute_with(|| {
+                let nonce = ParachainStaking::proxy_nonce(staker.account_id);
+                let unbond_call = create_call_for_signed_schedule_nominator_unbond(
+                    &staker,
+                    nonce,
+                    AMOUNT_TO_UNBOND.with(|v| *v.borrow()),
+                );
+                assert_ok!(AvnProxy::proxy(Origin::signed(staker.relayer), unbond_call, None));
+
+                assert!(unbond_event_emitted(Staker::default().account_id));
```

Comment:
Is the logic of balance changes checked somewhere else? These tests only assert success and event emitted.